### PR TITLE
Update libmodem doc 2.1.0

### PR DESCRIPTION
**Pull request by eivindj-nordic**

### CODE REVIEW

File: nrf_modem/doc/full_dfu.rst [hunk 1]
```diff
@@ -122,8 +122,10 @@ Firmware segments are written by using the following function call:
 	int nrf_modem_full_dfu_fw_write(uint32_t addr, uint32_t len, void *src)
 
 The modem library buffers the data with the same destination address, until one of the following conditions occur:
+
 * The buffered data reaches 8kb.
 * The destination address changes.
+
```
[the remark this answers is not in the document]
Comment (eivindj-nordic) [reply nrf_modem/doc/full_dfu.rst:128]: That is correct. https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/nrf_modem/doc/full_dfu.html

File: nrf_modem/doc/modem_trace.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+   :alt: Modem library trace processing
+
+   Modem library trace processing
+
+.. note::
+
+	If no trace data is available, :c:func:`nrf_modem_os_trace_get` will not return before new data becomes available or the modem is shut down or has faulted.
+	This require the traces to be processed in its own thread separate from the rest of the application.
+
+Asynchronous trace processing
```
Comment (lemrey) [inline nrf_modem/doc/modem_trace.rst:37]: I would rephrase this chapter a bit, and I am not sure about the chapter's tile.
The important information in this chapter is how to use `nrf_modem_trace_processed_callback`

I would say:

As the application processes the trace data, the application must call the `nrf_modem_trace_processed_callback` function to inform the Modem library that the trace data has been handled and can be freed. It is not necessary that the application has processed a whole "trace fragment" before calling `nrf_modem_trace_processed_callback`. To maximize the availability of the shared memory in the trace region, the application should call `nrf_modem_trace_processed_callback` as soon as some bytes are processed, as often as possible.

**Review by lemrey — COMMENTED** [review]
A problem with the MSCs is that they refer to a "trace backend" which is an NCS concept.
Here, we should just have the application, the Modem library and the Modem itself.

**Comment by lemrey** [comment]
Also the function was renamed to `nrf_modem_trace_processed`

File: nrf_modem/doc/modem_trace.rst (hunk 2 above)
Comment (eivindj-nordic) [reply nrf_modem/doc/modem_trace.rst:37]: I agree on the title, perhaps `Freeing processed traces` can be a better title?

I will make the other changes as well.

File: nrf_modem/doc/modem_trace.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   :alt: Modem library trace processing
+
+   Modem library trace processing
+
+.. note::
+
+	If no trace data is available, :c:func:`nrf_modem_os_trace_get` will block until new data becomes available or the modem is shut down or has faulted.
+	This require the traces to be processed in its own thread separate from the rest of the application.
+
+Freeing processed traces
```
Comment (divipillai) [inline nrf_modem/doc/modem_trace.rst:38]: **Processing modem traces** could be used.

File: nrf_modem/doc/modem_trace.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+.. note::
+
+	If no trace data is available, the :c:func:`nrf_modem_os_trace_get` function blocks until new data becomes available or the modem is shut down or faulted.
+	This require the traces to be processed in its own thread separate from the rest of the application.
+
+Processing modem traces
+***********************
+
+The application must call the: c:func: `nrf_modem_trace_processed` function as it processes the trace data to inform the Modem library that the trace data has been handled and can be released.
```
Comment (divipillai) [inline nrf_modem/doc/modem_trace.rst:41]:
```suggestion
The application must call the: c:func:`nrf_modem_trace_processed` function as it processes the trace data to inform the Modem library that the trace data has been handled and can be released.
```

c:func:`nrf_modem_trace_processed` - Some of the modem trace functions don't link. I couldn't find the doxyegn details of the modem trace function in any of the header files. Could we also add the modem trace functions to the API documentation (doxygen details) so that they are linked accordingly?

File: nrf_modem/doc/CHANGELOG.rst [hunk 5]
```diff
@@ -9,6 +9,20 @@ Changelog
...
+* Improvements to AT filters.
+  AT filters now support variable arguments.
+  The :c:member:`paused` is added to the :c:struct:`nrf_modem_at_cmd_filter` structure to pause filters whenever required.
+* Restructured the handling of modem traces.
+  The :c:func:`nrf_modem_os_trace_put` function that was called by the library is replaced with the :c:func:`nrf_modem_trace_get` function that must be called by the application.
+* Minor improvements to networking sockets.
+  :c:func:`nrf_recv` function with the ``NRF_MSG_WAITALL`` flag now returns if the socket is closed.
+  :c:func:`nrf_connect` function now returns an error if the socket is closed.
+
+.. TODO
```
[the remark this answers is not in the document]
Comment (eivindj-nordic) [reply nrf_modem/doc/CHANGELOG.rst:24]: There will most likely be more entries when opening the PR to NCS. However, I can remove the TODO for this PR.

File: nrf_modem/doc/modem_trace.rst (hunk 4 above)
Comment (eivindj-nordic) [reply nrf_modem/doc/modem_trace.rst:41]: I guess the reason why the files don't link is because they aren't exported to nrfxlib before the release. I'll add a temporary commit with new headers here.
